File: Samples/objective-c/ShowiFrameExample/ShowiFrameExample/ViewController.m

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UITextField *skuField;

@property (weak, nonatomic) IBOutlet CappasityModelView *modelView;

@property CappasityModel *model;

@property UIActivityIndicatorView *activityIndicator;

@end

@implementation ViewController

-(BOOL)shouldAutorotate {
    return false;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
    [_activityIndicator setCenter:CGPointMake(self.view.frame.size.width/2, self.view.frame.size.height/2)];
    [self.view addSubview:_activityIndicator];
    
    _modelView.delegate = self;
    _modelView.viewController = self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    [_modelView stopLoading];
}

- (IBAction)findButtonTapped:(UIButton *)sender {
    NSString* sku = [_skuField text];
    if (sku != nil && [sku length] > 0) {
        [_activityIndicator startAnimating];
        
        _model = [[CappasityModel alloc] initWithSku:sku delegate:self];
        [_model receiveInfo];
    }
}

- (NSString * _Nonnull)replace: (NSString * _Nonnull)embedCode :(NSString * _Nonnull) substring :(NSString * _Nonnull) value {
    return [embedCode stringByReplacingOccurrencesOfString:substring withString:value];
}

//MARK: - CappasityModelDelegate method
- (void)capServiceWithDidReceive:(NSInteger)error description:(NSString * _Nonnull)description userInfo:(id _Nullable)userInfo {
    [_activityIndicator stopAnimating];
    
    UIAlertController* alert = [UIAlertController alertControllerWithTitle:@"Error"
                                                                   message:description
                                                            preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction* defaultAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault
                                                          handler:^(UIAlertAction * action) {}];
    
    [alert addAction:defaultAction];
    [self presentViewController:alert animated:YES completion:nil];
}

//MARK: - CappasityModelDelegate method
- (void)capServiceDidReceiveModelInfo {
    CappasityModelViewParams *params = [[CappasityModelViewParams alloc] init];
    params.autoRun = true;
    [_modelView setWithModel:_model params:params];
}

//MARK: - CappasityModelViewDelegate method
- (void)capServiceWithDidReceive:(UIImage * _Nonnull)preview userInfo:(id _Nullable)userInfo {
    
}

//MARK: - CappasityModelViewDelegate method
- (void)capServiceDidLoadContent {
    [_activityIndicator stopAnimating];
}

@end

File: Samples/swift/ShowiFrameExample/ShowiFrameExample/ViewController.swift

import UIKit
import Cappasity

class ViewController: UIViewController {
    
    @IBOutlet weak var skuField: UITextField!
    @IBOutlet weak var modelView: CappasityModelView!
    
    fileprivate lazy var activityIndicator: UIActivityIndicatorView = {
        let activityIndicator = UIActivityIndicatorView()
        activityIndicator.frame = CGRect(x: 0, y: 0, width: 40, height: 40)
        activityIndicator.style = .gray

        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
        
        self.view.addSubview(activityIndicator)
        self.view.bringSubviewToFront(activityIndicator)
        
        NSLayoutConstraint(item: activityIndicator, attribute: .centerX, relatedBy: .equal, toItem: self.view, attribute: .centerX, multiplier: 1.0, constant: 0.0).isActive = true
        NSLayoutConstraint(item: activityIndicator, attribute: .centerY, relatedBy: .equal, toItem: self.view, attribute: .centerY, multiplier: 1.0, constant: 0.0).isActive = true
        
        return activityIndicator
    }()

    fileprivate var model: CappasityModel?
    
    override var shouldAutorotate: Bool {
        return false
    }
    
    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
        return .portrait
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        CappasityService.set(userAlias: "cappasity") //enter user's alias
        
        self.skuField.placeholder = "Enter sku"
        
        self.modelView.delegate = self
        self.modelView.viewController = self
        
        initGestureRecognizer()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        self.modelView.stopLoading()
    }
    
    @IBAction func findButtonTapped(_ sender: UIButton) {
        if let sku = skuField?.text {
            self.model = CappasityModel(withSku: sku, delegate: self)
            self.model?.receiveInfo()
            self.activityIndicator.startAnimating()
        }
    }
    
    private func initGestureRecognizer() {
        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboard))
        self.view.addGestureRecognizer(tap)
    }
    
    @objc private func dismissKeyboard() {
        self.view.endEditing(true)
    }
}

extension ViewController: CappasityModelDelegate {
    
    //MARK: - CappasityModelDelegate method
    func capService(didReceive error: CappasityServiceError) {
        self.activityIndicator.stopAnimating()
        Message.alert(viewController: self, title: "Error", message: error.localizedDescription, handler: nil)
    }
    
    //MARK: - CappasityModelDelegate method
    func capServiceDidReceiveModelInfo() {
        if let model = self.model {
            self.modelView.set(model, isAutorun: true)
        }
    }
    
    //MARK: - CappasityModelDelegate method
    func capService(didReceive preview: UIImage, userInfo: AnyObject?) {
        
    }
}

extension ViewController: CappasityModelViewDelegate {
    
    //MARK: - CappasityModelViewDelegate method
    func capServiceDidLoadContent() {
        self.activityIndicator.stopAnimating()
    }
}
